lib/quiet_logger.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class QuietLogger
+  PATH_INFO = "PATH_INFO"
+
+  attr_reader :app, :options, :paths
+
+  def initialize(app, options = {})
+    @app = app
+    @options = options
+    @paths = Array(options[:paths])
+  end
+
+  def call(env)
+    if silence_request?(env)
+      logger.silence { app.call(env) }
+    else
+      app.call(env)
+    end
+  end
+
+  private
+
+  def silence_request?(env)
+    paths.any? { |path| path === env[PATH_INFO] } # rubocop:disable Style/CaseEquality
```

Comment:
Because `===` works if you pass a regexp like `%r{\A/resource/\d+/path/to/ignore\z}`

Reply:
Ah yeah, I knew about `===` for regex matches, I just didn't realise we intended that behaviour here.

Reply:
Tbf I copied this over and removed the regex matches from `paths:`. But i'll just leave it here anyway in case we want to support regex matches going forward